marc.py

```diff
@@ -589,7 +589,7 @@ def create_record(cls, work, annotator, force_create=False, integration=None):
         record = None
         existing_record = getattr(work, annotator.marc_cache_field)
         if existing_record and not force_create:
-            record = Record(data=existing_record.encode("utf8"), force_utf8=True)
+            record = Record(data=existing_record, force_utf8=True)
```

Comment:
You already mentioned this in your description, but this change and the one below are potentially a problem.

Reply:
Right, now the data in `marc_record` will always be a byte string so the encoding and decoding is removed.

Reply:
I'm suspicious of the idea that a MARC record can ever be a Unicode string; it's a binary format. So I'm not sure what that decode() was doing in the first place.

If we're generating some subset of MARC that can always be a Unicode string, then I think it's fine to store it as Unicode and avoid some complexity on the database side.

Reply:
MARC is a binary format, but one of the [specified character encodings](https://www.loc.gov/marc/specifications/speccharucs.html) for that binary format is Unicode (specifically UTF-8 encoded unicode). I believe the `force_utf8` flag we are passing when creating the record sets the [marc record to the unicode encoding](https://gitlab.com/pymarc/pymarc/-/blob/master/pymarc/record.py#L103-104). It is output as a byte string, since MARC is a binary format, but as the code is written I believe we [always a unicode record](https://gitlab.com/pymarc/pymarc/-/blob/master/pymarc/record.py#L387-390). So we could either store it as bytes as it is here or we could store it as a string.

I'm not sure which one is best. 
Pros: 
 - Storing it as bytes would allow us to support other MARC encodings in the future

Cons:
 - Requires a DB migration
 - Makes it hard to query against the marc record
 
 Another possibility would be to store it in the database in some intermediate form. It looks like pymarc [offers a json serialization](https://gitlab.com/pymarc/pymarc/-/blob/master/pymarc/record.py#L448) that can be [round-tripped](https://gitlab.com/pymarc/pymarc/-/blob/master/pymarc/marcjson.py#L63). We could potentially store MARC as JSON in the DB.

Reply:
I see! That simplifies things a lot. Let's start storing our MARC records in the database as Unicode. I think that will solve the problem for the medium term. I don't want to do that DB migration as part of the Python 3 conversion, and I'd rather not do it at all.

Reply:
I think if we decide to stick with Unicode, we don't have to do any more work until we (for whatever reason) need to store in the database a MARC record that can't be represented as Unicode.